File: tests/machine/x/scala/record_assign.scala
case class Counter(var n: Int)

object record_assign {
  def inc(c: Counter) = {
    c.n = c.n + 1
  }
  
  def main(args: Array[String]): Unit = {
    var c = Counter(n = 0)
    inc(c)
    println((c.n))
  }
}

File: tests/machine/x/go/record_assign.go
//go:build ignore

package main

import (
	"fmt"
)

type Counter struct {
	N int `json:"n"`
}

// line 3
func inc(c *Counter) {
	c.N = (c.N + 1)
}

var c Counter

func main() {
	c = Counter{0}
	inc(&c)
	fmt.Println(c.N)
}
